### ISSUE TITLE
Please add Outcome to super repo for 1.70 release

### ISSUE THREAD

**Issue by ned14** [boostorg/boost#231]
https://github.com/ned14/boost-outcome

**Comment by pdimov** [comment]
You need to transfer the repo to an owner of boostorg, who would then transfer it to boostorg.

**Comment by ned14** [comment]
Firstly, my apologies for taking so long to get back to you. I was in Yosemite Valley on vacation where my European mobile phone could not get a signal from weird US frequencies, and thus the two factor authentication for logging into Github could not work. I am now on the coast in San Simeon on my second leg of vacation, where I can receive SMS 2FA.

Secondly, a very Merry Christmas to you Peter, and thanks for I assume taking up the role of sending boost-outcome into boostorg.

boost-outcome is now in your ownership. My thanks to you in advance.

**Comment by pdimov** [comment]
Here it is: https://github.com/boostorg/outcome

You'll need a `develop` branch though. Without it, I can't add it as a submodule on the `develop` branch of the superproject. I could create one off your current `master`, but since your repo is updated by a script, this may not be enough.

**Comment by djarek** [comment]
Would it be possible for the auto-generated commits in https://github.com/boostorg/outcome to at least have a reference to the original commit in the source repository? Makes it easier to follow changes in git.

**Comment by ned14** [comment]
Cool, thanks for the transfer. I can't fix the develop branch issue until I return to Europe, I have only my phone with me, no computers. I agree the commit history needs to be improved, and I will do that. We also need to decide how to handle the documentation, whether generation from Hugo or Travis doing it and pushing prebuilt static HTML is best.

**Comment by ned14** [comment]
Ok, we now have a commit by commit exact copy of standalone, with master and develop branches replicated including commit history and a web link to the original on github.

**Comment by pdimov** [comment]
Submodule added to develop, first results are in: https://www.boost.org/development/tests/develop/developer/outcome.html

We'll need to figure out something for the docs though.

**Comment by ned14** [comment]
Docs can be seen at https://boostorg.github.io/outcome/

Would a simple git submodule of that into boostorg/website suffice?

It's weird that the constexpr tests are failing above. They pass just fine on https://travis-ci.org/boostorg/outcome/jobs/478534166. Maybe I'll add more, newer compilers, see if there is something surprising there.

**Comment by pdimov** [comment]
```
compile-c-c++ ..\..\bin.v2\libs\outcome\test\constexpr.test\msvc-14.1\debug\thre
ading-multi\tests\constexpr.obj
constexpr.cpp
test\tests\constexpr.cpp(45): error C2338: result<int> is a literal type!
```
and others on msvc-14.1.

**Comment by ned14** [comment]
Don't worry about the constexpr test, it fails because of your shiny new error code. I've removed the test here.

Can we disable msvc-1.40 on the regression matrix? It'll never, ever pass.

I was explicitly told not to place git submodules into boostorg repos. However, Jamfile could locally git submodule add?

**Comment by pdimov** [comment]
There's nothing in "the rules" that prevents you submoduling your own submodule, but the problem is that nobody's going to `git submodule update` it. :-)

**Comment by pdimov** [comment]
And, you'll still need to update the submodule (commit) somehow. Which is more or less the same amount of work as updating `doc/html` itself. As you already have a script... maybe you can just keep the generated docs in the repo and update them from the script.

**Comment by pdimov** [comment]
To disable msvc-14.0, just add a requirement for cxx14_constexpr, in addition to your current one for cxx14_variable_templates.

**Comment by ned14** [comment]
I thought there was a policy against pushing a bunch of static HTML when it's generated? Thanks for the tip regarding disabling msvc-14.0.

**Comment by pdimov** [comment]
Having doc/html in the primary repo isn't good because of pull requests against the doc source often not being reflected in the generated html, but since your primary repo isn't here so you'll be applying the pull requests there, this doesn't really apply. It will generate a bit of churn in the revision history, but there are many other Boost libraries that have static html too. So it's not unheard of, and in this case, seems like the most straightforward approach.

**Comment by pdimov** [comment]
This is the other error on msvc-14.1:
```
compile-c-c++ ..\..\bin.v2\libs\outcome\test\hooks.test\msvc-14.1\debug\threadin
g-multi\tests\hooks.obj
hooks.cpp
test\tests\hooks.cpp(72): error C2280: 'boost::outcome_v2::basic_result<R,hook_t
est::error_code,boost::outcome_v2::policy::fail_to_compile_observers>::basic_res
ult<int,void>(T &&,boost::outcome_v2::basic_result<R,hook_test::error_code,boost
::outcome_v2::policy::fail_to_compile_observers>::implicit_constructors_disabled
_tag)': attempting to reference a deleted function
        with
        [
            R=int,
            T=int
        ]
```
...
```
c:\boost-git\develop\libs\outcome\include\boost\outcome\config.hpp(251): error C
2065: 'value': undeclared identifier
c:\boost-git\develop\libs\outcome\include\boost\outcome\config.hpp(251): error C
2131: expression did not evaluate to a constant
c:\boost-git\develop\libs\outcome\include\boost\outcome\config.hpp(251): note: a
 non-constant (sub-)expression was encountered
```

**Comment by ned14** [comment]
Yeah I saw that failure as well. Outcome was used by the Visual C++ team to debug MSVC, it's part of their test suite. So various older point releases fail the test suite. Latest VS2017 works fine, you can see there is another msvc-14.1 runner where it all passes. Will failing tests be a problem for release?

**Comment by ned14** [comment]
I also see that GCC 8 is ICEing in the constexpr tests. That's a long standing open bug in GCC, open since GCC 7, I'm subscribed to it, but they can't seem to figure out a solution.

**Comment by pdimov** [comment]
> Latest VS2017 works fine

Not for me, I was on latest when I tried it, it updated today to even more latest, still fails with the same error.

**Comment by pdimov** [comment]
No, a failing test is not a problem for the release.

**Comment by pdimov** [comment]
Ah, it passes with cxxstd=17, fails with cxxstd=14.

**Comment by pdimov** [comment]
Yes, docs seem to work fine: https://www.boost.org/doc/libs/develop/libs/outcome/doc/html/index.html

**Comment by danieljames** [comment]
FWIW The documentation on the website is just taken from the snapshot on bintray. The documentation is generated by Rene's CI build scripts. You can check in generated html file if you want, it's a pain to manage, but whatever works.
